cmd/transcriber/call/tracks.go

```diff
@@ -138,6 +140,24 @@ func (t *Transcriber) processLiveTrack(track trackRemote, sessionID string, user
 			continue
 		}
 
+		// We ignore out of order packets as they would cause synchronization
+		// issues. In the future we may want to reorder them but that requires us to keep
+		// buffers and complicate the whole process.
+		if pkt.Timestamp < prevRTPTimestamp {
+			slog.Debug("out of order packet",
+				slog.Int("diff", int(pkt.Timestamp)-int(prevRTPTimestamp)),
+				slog.String("trackID", ctx.trackID))
+
+			// Check that timestamp hasn't wrapped around. Fairly unlikely but it's
+			// a possibility since the starting timestamp is generated randomly so
+			// it could be close to the end of the uint32 range.
+			if hasWrappedAround := math.MaxUint32-prevRTPTimestamp < rtpTSWrapAroundThreshold; !hasWrappedAround {
+				continue
+			}
+
+			slog.Debug("ts wrap around detected", slog.String("trackID", ctx.trackID))
+		}
```

Comment:
Could you clarify what's happening here? It looks like if there is /not/ a wraparound we skip the packet?
Ahh... if it hasn't wrapped around, then we have an out of order and we skip it. If it has wrapped around, we continue. Could you maybe add a comment right at line 154 or 155 for future me who will be confused about the logic here? Thanks!

Reply:
Sure. Since a wraparound would look exactly like an out of order we need to differentiate, up to a threshold. So if it's not wraparound we continue (skip packet) and if it is it's all good as the difference is taken care of automatically by the `uint32` conversion. That's what the included test is checking.